# Practica 1 - Limpieza de datos
## Grupo 001 - Lunes 7-10 PM
## Equipo 9
+ 1842161 Torres Guerrero Saul Ángel
+ 1844656 Maldonado Ramirez Alejandra
+ 1857712 Morales Tovar Carlos de Jesús
+ 1941500 Espinosa Orrante Jesús Alejandro
+ 1857810 Rivera Castillo Saúl Andrés

## 1) Título de la base de datos: Wine Reviews

a) Nombre con el que se encuentra en la página encontrada: Wine Reviews

b) Url de la página: https://www.kaggle.com/zynicide/wine-reviews?select=winemag-data-130k-v2.csv 


## 2) Limpieza de datos

In [3]:
# Importar librerías y base de datos
import pandas as pd
import numpy as np
import json

wine=pd.read_csv("winemag-data-130k-v2.csv")

In [4]:
# Visualizar los datos
wine.shape

(129971, 14)

Nuestra base de datos está compuesta por 129,971 filas y 14 columnas

In [ ]:
# Descripción básica de columnas (tipo, elementos, etc.)
wine.columns

In [ ]:
wine.dtypes

Identificamos los nombres individuales de cada una de las 14 columnas para futuras referencias, de igual manera conocemos los tipos de datos que contienen

In [ ]:
# Cambiar nombres de columnas.
wine.rename(columns = {'taster_twitter_handle':'twitter'},inplace=True)
wine.drop('Unnamed: 0',axis = 1,inplace = True)
wine.columns

In [ ]:
wine['twitter'].dtype

In [ ]:
wine.head(3)

Modificamos el nombre de una columna solo para hacerlo más corto y eliminamos otra columna que era innecesaria

In [ ]:
# Guardar archivos en agrupaciones.
wine.country.unique()

In [17]:
EUA = wine[wine['country']=='US']

EUA.to_csv('EUA_Vinos.csv', index=False)

Guardamos en un archivo csv la info. de todos los vinos registrados como originarios de Estados Unidos

In [ ]:
country_vs_price =wine.loc[wine.country == 'US'].price
country_vs_price

Guardamos en una variable los precios de todos los vinos registrados como originarios de Estados Unidos

**PREGUNTA DE INTERÉS:**

*¿Para qué me puede servir estas agrupaciones para la práctica siguiente?*

Agrupar los datos es útil para hacer un análisis mas específico en caso de necesitarlo, en este caso de los vinos de EU y también únicamente de sus precios

In [ ]:
# Descripción básica de columnas (celdas vacias, etc.)
wine.isnull()

In [ ]:
wine.isnull().any()

In [ ]:
wine.isnull().sum()

Pudimos darnos cuenta de que tenemos muchas celdas vacías en distintas columnas de la base de datos

**PREGUNTA DE INTERÉS:**

*Importancia del analisis previo de los datos antes de iniciar las modificaciones.*

Asegurarnos de que las modificaciones sean necesarias, de lo contrario podríamos estar eliminando información relevante

In [29]:
wine_0 = wine.fillna(0)

In [ ]:
wine_0.isnull().any()

In [ ]:
wine_0.head(3)

Realizamos una copia de nuestros datos (wine_0) en la cual todos los valores con celdas nulas se llenaron con ceros

In [ ]:
#Extra
wine_extra = wine.replace('\@','',regex=True).astype(object)
wine_extra.head(5)

Eliminamos el carácter "@" de los datos de la columna twitter (de la base de datos original), podría considerarse innecesario

In [ ]:
wine_extra.dropna(thresh=int(wine_extra.shape[0] * .9), axis=1)

Se eliminaron la columnas que no cumplieran con tener el 90% o más de datos disponibles